api/dao/containerstorage.py

```diff
@@ -429,3 +435,162 @@ def get_all_for_targets(self, target_type, target_ids,
         if collection_id:
             query['collections'] = collection_id
         return self.get_all_el(query, user, projection)
+
+
+class AnalysisStorage(ContainerStorage):
+    def __init__(self):
+        super(AnalysisStorage, self).__init__('analyses', use_object_id=True)
+
+
+    def get_parent(self, parent_type, parent_id):
+        parent_storage = ContainerStorage.factory(parent_type)
+        return parent_storage.get_container(parent_id)
+
+
+    def get_analyses(self, parent_type, parent_id, inflate_job_info=False):
+        parent_type = containerutil.singularize(parent_type)
+        parent_id = bson.objectid.ObjectId(parent_id)
+        analyses = self.get_all_el({'parent.type': parent_type, 'parent.id': parent_id}, None, None)
+        if inflate_job_info:
+            for analysis in analyses:
+                self.inflate_job_info(analysis)
+        return analyses
+
+
+    def get_fileinfo(self, _id, filename=None):
```

Comment:
I wonder why there was a specific method created for this rather than just getting the analysis and reading it's file list. It's only used in one location so I would be in favor of removing it and moving the parts of it we need to where it is called a la `AnalysisStorage.get_container(<>).get('files', [])` or if that's too wordy writing a generic `get_files_for_container()` or something on `ContainerStorage`.